presto-product-tests/src/main/java/io/prestosql/tests/hive/util/TableLocationUtils.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests.hive.util;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.prestosql.tests.utils.QueryExecutors.onPresto;
+import static java.lang.String.format;
+
+public final class TableLocationUtils
+{
+    private TableLocationUtils()
+    {
+    }
+
+    public static String getTableLocation(String tableName, int partitionColumns)
+    {
+        StringBuilder regex = new StringBuilder("/[^/]*$");
+        for (int i = 0; i < partitionColumns; i++) {
+            regex.insert(0, "/[^/]*");
+        }
+        return getOnlyElement(onPresto().executeQuery(format("SELECT DISTINCT regexp_replace(\"$path\", '%s', '') FROM %s", regex.toString(), tableName)).column(1));
+    }
+
+    public static String getTablePath(String tableName, int partitionColumns)
+    {
+        String location = getTableLocation(tableName, partitionColumns);
+        try {
+            URI uri = new URI(location);
+            return uri.getPath();
+        }
+        catch (URISyntaxException e) {
```

Comment:
Is there any case where it's okay to return invalid URI? Isn't it better to reporting an exception properly instead of just returning the original table location?

Reply:
+1.
Can you provide explanation in the comment when do we expect the exception to happen.

Reply:
I did not meant that, I don't know how that slipped through.

Reply:
I guess this relates to https://github.com/prestosql/tempto/commit/706f495e2bee5886d8c7ad4ecbccd0d8d4eceee1

There, it was done defensively, as location can be specified as "path", not uri. Wasn't sure if metastore will always normalize it, by adding `hdfs://host/` prefix.
Maybe it does.

Reply:
Even if it doesn't, `new URL("/user/hive/warehouse/foo").getPath()` works just like `new URL("hdfs://hadoop-master:9000/user/hive/warehouse/foo").getPath()`.